### PR TITLE
ci: Upgrade most actions

### PR DESCRIPTION
**Pull request by zecakeh**
Should get rid of most warnings in CI.

I took the liberty to replace 2 unmaintained actions, these commits can be reverted if that's not wanted.

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -111,7 +111,7 @@ jobs:
           SLIDING_SYNC_PROXY_URL: "http://localhost:8118"
 
       - name: Upload to codecov.io
-        uses: codecov/codecov-action@v4
+         codecov/codecov-action@v4
```
Comment (poljar) [inline .github/workflows/coverage.yml:114]: v4 doesn't support tokenless uploads anymore: https://github.com/codecov/codecov-action?tab=readme-ov-file#breaking-changes.

I added the token using the `CODECOV_TOKEN` name as described in: https://docs.codecov.com/docs/adding-the-codecov-token#github-actions.

Could you please add it as described over here: https://github.com/codecov/codecov-action?tab=readme-ov-file#usage.

**Review by poljar — CHANGES_REQUESTED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (zecakeh) [reply .github/workflows/coverage.yml:114]: Wait I didn't actually change the action version. But strangely the `uses:` was removed here.

Comment (zecakeh) [reply .github/workflows/coverage.yml]: Fixed

Comment (poljar) [reply .github/workflows/coverage.yml]: Ah indeed.

**Comment by zecakeh** [comment]
Rebased on main for CI

**Review by poljar — APPROVED** [review]

**Pull request merged**